core/src/main/scala/org/broadinstitute/dsde/rawls/entities/base/InputExpressionReassembler.scala

```diff
@@ -169,17 +175,61 @@ object InputExpressionReassembler {
    */
   private def reconstructInputExprForEachEntity(mapOfEntityToEvaluatedExprMap: Map[EntityName, Try[Map[LookupExpression, JsValue]]],
                                                 parsedTree: ParseTree,
-                                                rootEntityNames: Option[Seq[EntityName]]): Map[EntityName, Try[JsValue]] = {
+                                                rootEntityNames: Option[Seq[EntityName]],
+                                                inputOption: Option[MethodInput]): Map[EntityName, Try[JsValue]] = {
+
+    def updateInputExprValueMapForObjectInput(inputExprWithEvaluatedRef: JsValue, objectFieldsFromInput: mutable.Seq[ValueTypeObjectFieldTypes]): JsObject = {
+      val updatedObject: Map[String, JsValue] = inputExprWithEvaluatedRef.asJsObject.fields.map { case (inputName, evaluatedValue) =>
+        val inputObjectField: Option[ValueTypeObjectFieldTypes] = objectFieldsFromInput.find(x => x.getFieldName == inputName)
+
+        val updatedValue: JsValue = inputObjectField match {
+          case Some(objectField) => objectField.getFieldType.getTypeName match {
+            case TypeNameEnum.OBJECT =>
+              // if there are nested objects, recursively call this method to reach leaf fields
```

Comment:
Do you have a test case that covers this? I'm having trouble understanding the test cases.

Reply:
Yes. [This](https://github.com/broadinstitute/rawls/blob/e7065edb153ba75f5fc47bdc436d12c67f11f15f/core/src/test/scala/org/broadinstitute/dsde/rawls/entities/local/LocalEntityProviderSpec.scala#L121-L131) test case should cover it

Reply:
Sounds good. I will add more permutations. I think multi-element array tests are already present. I will add the rest.